### PR TITLE
More gen 15 - settings gen

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] Settings gen
- [x] Test all settings. Seems fine
This one has some stuff that was typed incorrectly. i didn't touch it all cause i was afraid of breakage

### CODE REVIEW

File: shared/actions/settings.js [hunk 1]
```diff
@@ -182,42 +111,34 @@ function* toggleNotificationsSaga(): Saga.SagaGenerator<any, any> {
...
-        payload: undefined,
-      }: Constants.InvitesReclaimed)
-    )
+    yield Saga.put(SettingsGen.createInvitesReclaimed())
   } catch (e) {
     console.warn('Error reclaiming an invite:', e)
     yield Saga.put(
-      ({
-        type: Constants.invitesReclaimed,
-        payload: {errorText: e.desc + e.name, errorObj: e},
```
Comment (chrisnojima) [inline shared/actions/settings.js]: errorObj wasn't used

File: shared/actions/settings.js [hunk 2]
```diff
@@ -263,18 +184,21 @@ function* refreshInvitesSaga(): Saga.SagaGenerator<any, any> {
...
-  yield Saga.put({
-    type: Constants.invitesRefreshed,
-    payload: {
-      acceptedInvites,
-      pendingInvites,
-    },
-  })
+  yield Saga.put(
+    // $FlowIssues the typing of this is very incorrect. acceptedInvites shape doens't look anything like what we're pushing
+    SettingsGen.createInvitesRefreshed({
```
Comment (chrisnojima) [inline shared/actions/settings.js]: yeah this isn't typed correctly

File: shared/actions/settings.js [hunk 3]
```diff
@@ -291,10 +215,12 @@ function* sendInviteSaga(invitesSendAction: Constants.InvitesSend): Saga.SagaGen
       const invitationId = parsedBody.invitation_id.slice(0, 10)
       const link = 'keybase.io/inv/' + invitationId
       yield Saga.put(
-        ({
-          type: Constants.invitesSent,
-          payload: {email, invitationId},
-        }: Constants.InvitesSent)
+        SettingsGen.createInvitesSent()
+        // {
```
Comment (chrisnojima) [inline shared/actions/settings.js]: params not used at all

File: shared/actions/settings.js [hunk 4]
```diff
@@ -311,29 +237,25 @@ function* sendInviteSaga(invitesSendAction: Constants.InvitesSend): Saga.SagaGen
...
     yield Saga.call(delay, 500)
-    yield Saga.put({
-      type: Constants.notificationsRefreshed,
-      payload: {
-        groups: null,
-      },
-    })
+    yield Saga.put(
+      // $FlowIssue this isn't type correct at all TODO
+      SettingsGen.createNotificationsRefreshed({
```
Comment (chrisnojima) [inline shared/actions/settings.js]: same here, typing isn't correct

File: shared/settings/invites/container.js [hunk 5]
```diff
@@ -1,22 +1,9 @@
...
-import Invites, {type Props, type PendingInvite} from '.'
-import * as Constants from '../../constants/settings'
-import {invitesReclaim, invitesRefresh, invitesSend} from '../../actions/settings'
+import * as SettingsGen from '../../actions/settings-gen'
+import Invites, {type PendingInvite} from '.'
 import {openURLWithHelper} from '../../util/open-url'
 import {navigateAppend} from '../../actions/route-tree'
-import {connect, type TypedState} from '../../util/container'
-
-// TODO recompose this
```
Comment (chrisnojima) [inline shared/settings/invites/container.js]: might as well do this now

File: shared/actions/settings.js (hunk 1 above)
Comment (buoyad) [reply shared/actions/settings.js]: is the action used at all?

Comment (chrisnojima) [reply shared/actions/settings.js]: not really

**Pull request merged**